### ISSUE TITLE
Apply Existing Snapshot Limit to Auto Export Directory

### ISSUE THREAD

**Issue by davidleejy** [mbnuqw/sidebery#1411]
## Motivation & Description

The ability of Sidebery to automatically create and export snapshots is essentially a life-saving **backup** when a machine with 100s of opened tabs suddenly fails – especially when snapshots are exported to a remote/cloud-synced location*, e.g., Dropbox, Google Drive, OneDrive.

<img width="472" alt="Screenshot 2024-01-11 at 2 05 05 PM" src="https://github.com/mbnuqw/sidebery/assets/1690072/27971371-83f6-451a-a741-bdca09360aa6">

However, the snapshot limit (circled green in screenshot below) only applies to snapshots viewable in Sidebery's native snapshot viewer, but not to the auto export directory (circled pink in screenshot below).

As a result, the auto export directory grows without a limit. It would be nice if the existing snapshot limit could also be applied to the Auto Export directory.

<img width="681" alt="Screenshot 2024-01-11 at 1 49 23 PM" src="https://github.com/mbnuqw/sidebery/assets/1690072/8618bd08-4445-4968-9aa0-355e98c60992">

## Footnotes

\* I do this by symbolically linking a directory in the Downloads dir to a Dropbox dir:

```bash
ln -s /Users/d/Dropbox/Sidebery/snapshots /Users/d/Downloads/sidebery-snapshots-do-not-delete
```

**Comment by Heyian** [comment]
For anyone who is stumbling on this issue just like me, are on Windows and would like to automate the cleanup of the folder, I made a daily Task with the Task Scheduler to clean files older than 15 days in the folder. Here's how I did it :

1. Open Task Scheduler and make a new task, I named mine "Delete old Sideberry exports"

2. In the New Action window, specify the following data:

    Action: Start a program
    Program/script: `ForFiles.exe`
    Add arguments (optional): `/p "C:\Users\{user}\Downloads\Sidebery" /s /d -15 /c "cmd /c del @file"`
    Even though Windows says it's optional, it is not for our case, don't skip this.
    ### Don't forget to change the location to **_YOUR_** Sideberry export folder

3. Set the Trigger:
        Go to the Triggers tab and click on the New button.
        Under the Begin task, select a schedule in the drop-down list, I chose Daily.

4. Configure Settings:
        Switch to the Settings tab.
        Enable "Allow task to be run on demand" and "Run task as soon as possible after a scheduled start is missed".
        Click OK to create your task.

And that's it, it nows cleans up the folder everyday, it's super fast and I don't expect much hiccup with this method.

Don't forget to test it before simply letting it do it's job ;)

**Comment by Heyian** [comment]
Also, I think this is the same as Issue #1294

**Comment by davidleejy** [comment]
Thanks Heyian for sharing your method for managing the number of snapshots backed-up. 😄 

Re-examining this problem two months after opening this issue, I suspect that this problem of Sidebery not natively providing an option to delete older snapshots **isn't serious** for many users primarily because snapshot files are quite small relative to the hard drive memory common in modern entry-level computers. Currently, I have between 500 to 800 tabs opened in Firefox and all of them can be saved to a single snapshot file (as .json, or .md format) sized ≤400kB.

I have not implemented any automation to clean up snapshots (e.g., task scheduler, chron). I've configured Sidebery to save snapshots once every three days – resulting in 10 snapshots per month. Having configured each snapshot to be saved in two file formats – md & json – and with each file occupying fewer than 400kB (thus ≤800kB for two files), my Sidebery snapshot back-ups consume _at most_ an additional 8mB per month, or _at most_ 96mB annually. At this rate, I think it'd all right to manually house-keep the Sidebery snapshots directory once annually or even once every three years. 

I'm writing to share my thoughts because I don't wish to alarm other Sidebery users, especially those new to Sidebery, to cause them to eschew Sidebery 😆

_______

**A Related Issue: How to Automatically Back-up Sidebery Snapshots onto a Cloud**

A bit of a digression from the issue here, but I figured that people looking at backing up their snapshots might also be considering about placing their snapshots on a cloud (e.g., dropbox, microsoft onedrive, google drive), and as such, might benefit from reading about the steps I took to accomplish this which I document in detail in [discussion #1497](https://github.com/mbnuqw/sidebery/discussions/1497)
